source/4_tools/regina/src/regina.cpp

```diff
@@ -1,54 +1,865 @@
+            renderer::commands::ConstantLayoutDescription {renderer::commands::ConstantType::CBUFFER, "PassConstants", 1}
+        };
+        g_regina_ctx.shader_program = renderer::link_shader(rsl::move(link_shader_command_desc));
+
+        // Input Layout
+        renderer::commands::CreateInputLayoutCommandDesc input_layout_command_desc;
+        input_layout_command_desc.input_layout =
+        {
+            renderer::commands::InputLayoutDescription { "POSITION", 0, renderer::VertexBufferFormat::FLOAT3, 0, 0, renderer::InputLayoutClassification::PER_VERTEX_DATA, 0 },
+            renderer::commands::InputLayoutDescription { "COLOR", 0, renderer::VertexBufferFormat::FLOAT4, 0, 12, renderer::InputLayoutClassification::PER_VERTEX_DATA, 0 }
+        };
+        g_regina_ctx.input_layout = renderer::create_input_layout(rsl::move(input_layout_command_desc));
+
+        return true;
+    }
+
+    //-------------------------------------------------------------------------
+    bool build_cube_geometry()
+    {
+        auto box = mesh_factory::create_box<u16>(1.5f, 0.5f, 1.5f, 0);
+
+        auto total_vertex_count = box.vertices().size();
+        auto total_index_count = box.indices().size();
+
+        rsl::vector<renderer::VertexPosCol> box_vertices((rsl::Capacity)total_vertex_count);
+        for (const mesh_factory::Vertex& v : box.vertices())
+        {
+            renderer::VertexPosCol nv({ v.position.x, v.position.y, v.position.z }, { v.position.x, v.position.y, v.position.z, 1.0f });
+            box_vertices.push_back(nv);
+        }
+
+        rsl::vector<u16> box_indices((rsl::Capacity)total_index_count);
+        box_indices.insert(box_indices.end(), rsl::begin(box.indices()), rsl::end(box.indices()));
+
+        const u32 vb_byte_size = total_vertex_count * sizeof(renderer::VertexPosCol);
+        const u32 ib_byte_size = total_index_count * sizeof(u16);
+
+        g_regina_ctx.mesh_cube = rsl::make_unique<renderer::Mesh>();
+        g_regina_ctx.mesh_cube->name = rsl::medium_stack_string("box_geometry");
+
+        renderer::commands::CreateBufferCommandDesc v_create_buffer_command_desc = create_buffer_parameters<renderer::VertexPosCol>(box_vertices.data(), box_vertices.size());
+        g_regina_ctx.mesh_cube->vertex_buffer = renderer::create_vertex_buffer(rsl::move(v_create_buffer_command_desc));
+        renderer::commands::CreateBufferCommandDesc i_create_buffer_command_desc = create_buffer_parameters<u16>(box_indices.data(), box_indices.size());
+        g_regina_ctx.mesh_cube->index_buffer = renderer::create_index_buffer(rsl::move(i_create_buffer_command_desc));
+
+        g_regina_ctx.mesh_cube->vertex_byte_stride = sizeof(renderer::VertexPosCol);
+        g_regina_ctx.mesh_cube->vertex_buffer_byte_size = vb_byte_size;
+        g_regina_ctx.mesh_cube->index_format = renderer::IndexBufferFormat::R16_UINT;
+        g_regina_ctx.mesh_cube->index_buffer_byte_size = ib_byte_size;
+
+        renderer::Submesh submesh;
```

Comment:
This is the early stage to setup static batching within the renderer. Currently these are all parts of the same geometry but that is not what is going to happen in the future a glimps can already be spotted in the scene_setup I provided. there is one big chunk of geometry with multiple submeshes. This type of architecture will eventually result in having 1 drawcall for all the sets of geometries within the scene ( with the same material ) instead of having draw calls for every set of geometry loaded into memory. Another thing to point out is that this is how every major format (fbx, gltf, stl, ... ) handles geometry within geometry. yes this is manually as the data I use now is also calculated from code instead of being imported but when we read geometry in from disk values will be retrieved from those files. I don't understand why this is a big deal. Aside from adding a bit of syntax sugar this is how I want it to work.